pulsar-common/src/main/java/org/apache/pulsar/common/util/ObjectMapperFactory.java

```diff
@@ -109,7 +112,10 @@
 @Slf4j
 public class ObjectMapperFactory {
     public static ObjectMapper create() {
-        ObjectMapper mapper = new ObjectMapper();
+        ObjectMapper mapper = new ObjectMapper()
+                .registerModule(new ParameterNamesModule())
```

Comment:
Do we use it somewhere ? I think to work javac needs to be run with  the `-parameters` option

Reply:
The point of this is to allow using parameter names in your own Java classes without any special config.

Reply:
You mean for plugin/connector/function developers ?

Reply:
yes, also for applications where the Pulsar client is used. It's been a big mistake in Pulsar initially to not enable https://github.com/FasterXML/jackson-modules-java8 modules by default. Customizing Jackson configuration is possible, but it isn't properly configured. It doesn't work with the shaded client.